flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/sql/ml/SqlMLEvaluateTableFunction.java

```diff
@@ -0,0 +1,208 @@
+        final RelDataTypeFactory typeFactory = opBinding.getTypeFactory();
+        final RelDataType inputRowType = opBinding.getOperandType(0);
+
+        // Create a MAP type for evaluation results
+        RelDataType keyType = typeFactory.createSqlType(SqlTypeName.VARCHAR);
+        RelDataType valueType = typeFactory.createSqlType(SqlTypeName.DOUBLE);
+        RelDataType mapType = typeFactory.createMapType(keyType, valueType);
+
+        return typeFactory
+                .builder()
+                .kind(inputRowType.getStructKind())
+                .add("result", mapType)
+                .build();
+    }
+
+    private static class EvaluateOperandMetadata implements SqlOperandMetadata {
+        private static final List<String> PARAM_NAMES =
+                List.of(
+                        PARAM_INPUT,
+                        PARAM_MODEL,
+                        PARAM_LABEL,
+                        PARAM_FEATURE,
+                        PARAM_TASK,
+                        PARAM_CONFIG);
+        private static final List<String> MANDATORY_PARAM_NAMES =
+                List.of(PARAM_INPUT, PARAM_MODEL, PARAM_LABEL, PARAM_FEATURE);
+
+        EvaluateOperandMetadata() {}
+
+        @Override
+        public List<RelDataType> paramTypes(RelDataTypeFactory typeFactory) {
+            return Collections.nCopies(
+                    PARAM_NAMES.size(), typeFactory.createSqlType(SqlTypeName.ANY));
+        }
+
+        @Override
+        public List<String> paramNames() {
+            return PARAM_NAMES;
+        }
+
+        @Override
+        public boolean checkOperandTypes(SqlCallBinding callBinding, boolean throwOnFailure) {
+            if (!SqlValidatorUtils.checkTableAndDescriptorOperands(callBinding, 2, 3)) {
+                return SqlValidatorUtils.throwValidationSignatureErrorOrReturnFalse(
+                        callBinding, throwOnFailure);
+            }
+
+            if (!SqlValidatorUtils.throwExceptionOrReturnFalse(
+                    checkModelSignature(callBinding, 3, 2), throwOnFailure)) {
+                return false;
+            }
+
+            if (callBinding.getOperandCount() == PARAM_NAMES.size()) {
+                // Check task and config parameters
+                if (!SqlValidatorUtils.throwExceptionOrReturnFalse(
+                        checkTask(callBinding.operand(4)), throwOnFailure)) {
+                    return false;
+                }
+
+                if (!SqlValidatorUtils.throwExceptionOrReturnFalse(
+                        checkConfig(callBinding.operand(5)), throwOnFailure)) {
+                    return false;
+                }
+            }
+
+            if (callBinding.getOperandCount() == MANDATORY_PARAM_NAMES.size() + 1) {
+                // Last param can be config or task
+                return SqlValidatorUtils.throwExceptionOrReturnFalse(
+                        checkTaskOrConfig(callBinding.operand(4)), throwOnFailure);
```

Comment:
It's better we use `MANDATORY_PARAM_NAMES.size() + 1` here because if condition doesn't use hard-code number